components/pages/workPageLayout/LoanMaterialModal.tsx

```diff
@@ -0,0 +1,164 @@
+        {isLoadingCovers && (
+          <div className="aspect-1/1 h-36 lg:aspect-4/5">
+            <CoverPictureSkeleton />
+          </div>
+        )}
+        <CoverPicture
+          alt="Forsidebillede på værket"
+          lowResSrc={lowResCover || ""}
+          src={coverSrc?.[0] || ""}
+        />
+        <MaterialTypeIconWrapper
+          iconName={getManifestationMaterialTypeIcon(manifestation)}
+          className="bg-background absolute -bottom-6 h-10 w-10"
+        />
+      </div>
+
+      {/* Description */}
+      <p className="text-typo-body-lg mt-10 mb-5 w-full text-center">
+        {isLoadingLoans && (
+          <div className="bg-background-skeleton h-[26px] w-[500px] animate-pulse rounded-full" />
+        )}
+        {!canUserLoanMoreEMaterials(dataLoans, manifestation) && (
+          <>
+            Du kan desværre ikke låne flere{" "}
+            <span className="font-bold">
+              {`"${getManifestationMaterialTypeSpecific(manifestation)}er"`}
+            </span>{" "}
+            i denne måned.
+          </>
+        )}
+        {error && "Vi kunne desværre ikke låne materialet. Prøv igen senere."}
+        {isErrorLoans &&
+          "Der sket desværre et fejl ved at checke om du kan låne materialet. Prøv igen senere."}
+        {!error &&
+          !isLoadingLoans &&
+          canUserLoanMoreEMaterials(dataLoans, manifestation) &&
+          `Er du sikker på at du vil låne denne ${getManifestationMaterialTypeSpecific(manifestation) || "material"}?`}
+      </p>
+
+      <div className="flex flex-row items-center justify-center gap-6">
+        {/* Only show "approve loan" button if user can still loan more materials */}
+        {canUserLoanMoreEMaterials(dataLoans, manifestation) && !error && !isErrorLoans && (
+          <Button
+            theme={"primary"}
+            size={"lg"}
+            onClick={handleLoanMaterial}
+            disabled={isHandlingLoan || isLoadingLoans}>
+            {!isHandlingLoan && "Ja"}
+            {isHandlingLoan ||
+              (isLoadingLoans && (
+                <Icon
+                  name="go-spinner"
+                  ariaLabel="Indlæser"
+                  className="animate-spin-reverse h-[24px] w-[24px]"
+                />
+              ))}
+          </Button>
+        )}
+        <Button
+          size={"lg"}
+          onClick={() => {
+            setIsOpen(prev => !prev)
+          }}
+          disabled={isHandlingLoan || isLoadingLoans}>
+          {!canUserLoanMoreEMaterials(dataLoans, manifestation) || error || isErrorLoans
+            ? "Luk"
+            : "Nej"}
+          {isLoadingLoans && (
+            <Icon
+              name="go-spinner"
+              ariaLabel="Indlæser"
+              className="animate-spin-reverse h-[24px] w-[24px]"
+            />
+          )}
+        </Button>
+      </div>
+    </ResponsiveDialog>
```

Comment:
I think the component is getting too big and it would be worth splitting into smaller components. 
Maybe it is not worth doing before it is transformed into using the global modal system, but then we could write a todo about it. 
I think it is getting difficult to get an overview of what is happening when a component is getting this big. Especially also because there is som much logic checking. 
By splitting the different cases into sub components makes it easier to understand the logic because it stands out. And then you can dive into the sub components if you want to have a closer look into the rendering of the fragments.

Reply:
I am writing a todo and making a ticket